source/models/index.py

```diff
@@ -1,29 +1,33 @@
+from genericpath import isfile
 import os
 from outputs.exceptions import *
 
 class Index:
-    def __init__(self, table, index_value):
+    def __init__(self, table, index_name, index_value = None):
         self.table = table
```

Comment:
What is the difference between index name and index value ?

Reply:
index name is like ```first name``` folder
but index value is like ```joseph.txt``` file that hold the PKs of students has joseph name

Reply:
same answer